src/mirror.c

```diff
@@ -114,13 +114,6 @@ static int unset_mirror_url()
 		goto out;
 	}
 
-	/* we need to set the version and content URLs again if not they will
-	 * remain with the value from the mirror */
-	free_string(&version_url);
-	free_string(&content_url);
-	set_version_url(NULL);
```

Comment:
why are these not necessary anymore?

Reply:
On mirror --unset those variables weren't set because swupd isn't initialized yet. So we don't need/should unset them.
The only place we really need to have this is in handle_mirror_if_stale() function. I moved this to that function then.

Reply:
ahh make sense